receiver/statsdreceiver/receiver.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsdreceiver
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/protocol"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/transport"
+)
+
+var _ component.MetricsReceiver = (*statsdReceiver)(nil)
+
+// statsdReceiver implements the component.MetricsReceiver for StatsD protocol.
+type statsdReceiver struct {
+	sync.Mutex
+	logger *zap.Logger
+	config *Config
+
+	server       transport.Server
+	reporter     transport.Reporter
+	parser       protocol.Parser
+	nextConsumer consumer.MetricsConsumerOld
+
+	startOnce sync.Once
+	stopOnce  sync.Once
+}
+
+// New creates the StatsD receiver with the given parameters.
+func New(
+	logger *zap.Logger,
+	config Config,
+	nextConsumer consumer.MetricsConsumerOld) (component.MetricsReceiver, error) {
+	if nextConsumer == nil {
+		return nil, componenterror.ErrNilNextConsumer
+	}
+
+	if config.NetAddr.Endpoint == "" {
+		config.NetAddr.Endpoint = "localhost:8125"
+	}
+
+	server, err := transport.NewUDPServer(config.NetAddr.Endpoint)
```

Comment:
You might add a TODO or file an issue about support for UNIX socket support, which is fairly common (and should be little new code). I've read that TCP support is sometimes used, as well, but I don't know if that is common (and I would not recommend it). The UNIX socket support is something many Statsd agents also support. See https://github.com/open-telemetry/opentelemetry-go-contrib/blob/master/exporters/metric/dogstatsd/internal/statsd/conn.go